lib/index.js

```diff
@@ -211,20 +212,37 @@ Segment.prototype.ampId = function(ctx) {
 
 Segment.prototype.send = function(path, msg, fn) {
   var url = 'https://' + this.options.apiHost + path;
-  var headers = { 'Content-Type': 'text/plain' };
   fn = fn || noop;
   var self = this;
 
   // msg
   msg = this.normalize(msg);
 
   // send
-  send(url, msg, headers, function(err, res) {
-    self.debug('sent %O, received %O', msg, arguments);
-    if (err) return fn(err);
-    res.url = url;
-    fn(null, res);
-  });
+  if (this.options.beacon && navigator.sendBeacon) {
+    // Beacon returns false if the browser couldn't queue the data for transfer
+    // (e.g: the data was too big)
```

Comment:
Not sure, it varies between browsers. [Looks like Chrome's limit is 65536 characters.](https://stackoverflow.com/questions/28989640/navigator-sendbeacon-data-size-limits) I guess it really depends on how much custom data people send. This falls back to AJAX if the data is too big since `navigator.sendBeacon` will return false.